### DISCUSSION TITLE
Docker image should not run as root

### DISCUSSION THREAD

**Discussion by putnam** [open-webui/open-webui#1698] [General]
**Is your feature request related to a problem? Please describe.**
Docker containers running as root are dangerous. It is a best practice to write your Dockerfile such that it drops privileges to another user ID before starting up, or, alternatively, write it in such a way that the `--user` setting in Docker will work as expected. Currently the Dockerfile included with this project depends completely on the context of the root user and will not function if you try to run it any other way.

**Describe the solution you'd like**
Use either the privilege dropping strategy (preferably with a UID/GID environment variable as many other containers use) or support the --user param correctly.

**Comment by plattenschieber** [comment]
I faced similar issues when trying to run the container in OpenShift, which is quite more strict than base Kubernetes. E.g. containers generally start with a random user, which resulted in
- `ollama` not starting correctly as mount is expected under `/.ollama` 
    > Couldn't find '/.ollama/id_ed25519'. Generating new private key.
Error: could not create directory mkdir /.ollama: permission denied
- container failing to start as weirdly copying files during runtime (which fails, because container isn't root)
    > ```shutil.copyfile(f"{FRONTEND_BUILD_DIR}/favicon.png", f"{STATIC_DIR}/favicon.png")```

For now I just gave it a more powerful ServiceAccount to work with. But that ain't a solution ✌

Happy to contribute and make this more robust. Also please convert this back to an issue, as the deployment is not working as-is.

**Comment by putnam** [comment]
I think it's a meaningful security issue to even have this Dockerfile as-is (and likewise for the ollama/ollama image). Relegating it to a discussion doesn't seem appropriate. It's been standard practice to avoid running as root for years.

**Reply by justinh-rahb** [reply]
We await your PR 🙏

**Reply by Katharta** [reply]
Not everyone is a developer.

**Reply by dleske** [reply]
A [PR submitted for this](https://github.com/open-webui/open-webui/pull/14146) was closed without comment; the [discussion](https://github.com/open-webui/open-webui/discussions/14145) has no comments.
